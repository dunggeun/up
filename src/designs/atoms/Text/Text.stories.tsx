import React from 'react';
import { View } from 'dripsy';
import { Text } from './Text';

import type { ComponentProps } from 'react';
import type { Meta, StoryFn } from '@storybook/react';

export default {
  title: 'components/Text',
  component: Text,
} as Meta<typeof Text>;

type TextProps = ComponentProps<typeof Text>;
type TextStory = StoryFn<typeof Text>;

const Template = (args: TextProps) => (
  <View sx={{ flex: 1, padding: '$04' }}>
    <Text {...args} />
  </View>
);

export const Default: TextStory = (args: TextProps) => {
  return <Template {...args} />;
};

Default.args = {
  children: 'Text',
};
