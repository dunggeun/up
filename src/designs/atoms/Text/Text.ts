import { Text as DripsyText } from 'dripsy';

import type { ComponentPropsWithoutRef } from 'react';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface TextProps
  extends ComponentPropsWithoutRef<typeof DripsyText> {}

export const Text = DripsyText;
