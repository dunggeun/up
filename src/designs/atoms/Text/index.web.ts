import { Text as DripsyText } from 'dripsy';

export const Text = DripsyText;
