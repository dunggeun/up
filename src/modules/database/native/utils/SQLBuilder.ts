/* eslint-disable no-case-declarations */
import dedent from 'dedent';
import { Logger } from 'src/modules/logger';
import type { DatabaseRecord, Value, WhereConditions } from '../../types';

type QueryType = 'select' | 'count' | 'insert' | 'update' | 'delete';

const TAG = 'SQLBuilder';

export class SQLBuilder {
  private _tableName?: string;
  private _type?: QueryType;
  private _values?: DatabaseRecord;
  private _limit?: number;
  private _orderTarget?: string;
  private _by?: 'desc' | 'asc';
  private _condition?: WhereConditions;

  private stringify(value: Value): string {
    if (typeof value === 'string') {
      return `"${value}"`;
    } else if (value === null) {
      return 'null';
    }
    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
    return `${value}`;
  }

  table(tableName: string): this {
    this._tableName = tableName;
    return this;
  }

  type(type: QueryType): this {
    this._type = type;
    return this;
  }

  values(values: DatabaseRecord): this {
    this._values = values;
    return this;
  }

  where(condition: WhereConditions): this {
    this._condition = condition;
    return this;
  }

  limit(limit: number): this {
    this._limit = limit;
    return this;
  }

  orderBy(target: string, by: 'desc' | 'asc'): this {
    this._orderTarget = target;
    this._by = by;
    return this;
  }

  build(): string {
    if (!(this._type && this._tableName)) {
      throw Error('invalid query type or table name');
    }

    let query: string;
    const WHERE_CONDITION =
      this._condition &&
      Object.entries(this._condition)
        .map(([column, { symbol, value }]) => {
          return `${column} ${symbol} ${this.stringify(value)}`;
        })
        .join(' AND ');
    const WHERE_STATEMENTS = WHERE_CONDITION ? `WHERE ${WHERE_CONDITION}` : '';

    switch (this._type) {
      case 'select':
        const shouldOrdering = this._orderTarget && this._by;
        const ORDER_BY_STATEMENTS = shouldOrdering
          ? // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
            `ORDER BY ${this._orderTarget!} ${this._by!}`
          : '';

        const LIMIT_CONSTRAINT =
          this._limit !== undefined ? `LIMIT ${this._limit}` : '';

        query = dedent`
          SELECT * FROM ${this._tableName}
          ${WHERE_STATEMENTS}
          ${ORDER_BY_STATEMENTS}
          ${LIMIT_CONSTRAINT};
        `;
        break;

      case 'count':
        query = dedent`SELECT count(*) as count FROM ${this._tableName};`;
        break;

      case 'insert':
        const VALUE_COLUMNS = Object.keys(this._values ?? {}).join(',');
        const VALUES = Object.values(this._values ?? {})
          .map((value) => this.stringify(value))
          .join(',');

        query = dedent`
          INSERT INTO ${this._tableName} (${VALUE_COLUMNS})
          VALUES (${VALUES});
        `;
        break;

      case 'update':
        const SET_STATEMENTS = Object.entries(this._values ?? {})
          .map(([column, value]) => `${column} = ${this.stringify(value)}`)
          .join(',');

        query = dedent`
          UPDATE ${this._tableName}
          SET ${SET_STATEMENTS}
          ${WHERE_STATEMENTS};
        `;
        break;

      case 'delete':
        query = dedent`
          DELETE FROM ${this._tableName}
          ${WHERE_STATEMENTS};
        `;
        break;
    }

    query = query.replace(/[\r\n]/g, ' ');
    Logger.debug(TAG, query);

    return query;
  }
}
