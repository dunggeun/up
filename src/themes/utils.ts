import { Platform } from 'react-native';

export const webFont = (font: string): string => {
  return Platform.select({
    web: `${font}, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, Inter-serif`,
    default: font,
  });
};
