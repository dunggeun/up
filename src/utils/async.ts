export const delay = (duration: number): Promise<void> => {
  return new Promise((resolve) => {
    setTimeout(resolve, duration);
  });
};

/* istanbul ignore next */

export const debounce = <T>(
  callback: (...args: T[]) => void,
  // eslint-disable-next-line @typescript-eslint/no-shadow
  delay: number,
): ((...args: T[]) => void) => {
  let timer: NodeJS.Timeout;
  return (...args) => {
    clearTimeout(timer);
    timer = setTimeout(() => callback(...args), delay);
  };
};
