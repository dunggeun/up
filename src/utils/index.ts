export * from './async';
export * from './common';
export * from './locales';
