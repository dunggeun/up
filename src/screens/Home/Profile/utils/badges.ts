import { t } from 'src/translations';
import dummy from 'src/assets/badges/dummy.png';

import type { ImageSourcePropType } from 'react-native';

const BADGE_SET = [[t('badge.0'), dummy]] as const;
const FALLBACK_BADGE = ['', dummy] as const;

export const getBadge = (id: number): { title: string, image: ImageSourcePropType } => {
  const badge = BADGE_SET[id];
  const [title, image] = badge ?? FALLBACK_BADGE;
  return { title, image };
};
