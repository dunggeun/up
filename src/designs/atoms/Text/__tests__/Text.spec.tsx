import React from 'react';
import { render as testRender, cleanup } from '@testing-library/react-native';
import { withDripsy } from 'tests/utils';
import { Text } from '../Text';

import type { TextProps } from '../Text';

const render = (props: TextProps): ReturnType<typeof testRender> =>
  testRender(withDripsy(<Text {...props} />));

describe('atoms/Text', () => {
  const LABEL = 'Text';

  afterEach(() => {
    cleanup();
  });

  describe('렌더링 되었을 때', () => {  
    it('스냅샷이 일치해야 한다', () => {
      const tree = render({ children: LABEL }).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
