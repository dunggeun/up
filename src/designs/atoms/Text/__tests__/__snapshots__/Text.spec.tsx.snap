// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Text 렌더링 되었을 때 스냅샷이 일치해야 한다 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "TmoneyRoundWind-Regular",
        "fontSize": 16,
      },
      undefined,
      {},
    ]
  }
>
  Text
</Text>
`;
